v2/tests/log_context_property_type_ascii_char_ptr_ut/log_context_property_type_ascii_char_ptr_ut.c

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdarg.h>
+#include <string.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/log_context_property_type.h"
+#include "c_logging/log_context_property_type_if.h"
+
+#include "c_logging/log_context_property_type_ascii_char_ptr.h"
+
+static size_t asserts_failed = 0;
+
+// defines how many mock calls we can have
+#define MAX_MOCK_CALL_COUNT (128)
+
+#define MOCK_CALL_TYPE_VALUES \
+    MOCK_CALL_TYPE_snprintf, \
+    MOCK_CALL_TYPE_vsprintf, \
+    MOCK_CALL_TYPE_vsnprintf \
+
+MU_DEFINE_ENUM(MOCK_CALL_TYPE, MOCK_CALL_TYPE_VALUES)
+
+// very poor mans mocks :-(
+typedef struct snprintf_CALL_TAG
+{
+    bool override_result;
+    int call_result;
+    const char* format_arg;
+} snprintf_CALL;
+
+typedef struct vsprintf_CALL_TAG
+{
+    bool override_result;
+    int call_result;
+    const char* format_arg;
+} vsprintf_CALL;
+
+typedef struct vsnprintf_CALL_TAG
+{
+    bool override_result;
+    int call_result;
+    const char* format_arg;
+} vsnprintf_CALL;
+
+typedef struct MOCK_CALL_TAG
+{
+    MOCK_CALL_TYPE mock_call_type;
+    union
+    {
+        snprintf_CALL snprintf_call;
+        vsprintf_CALL vsprintf_call;
+        vsnprintf_CALL vsnprintf_call;
+    } u;
+} MOCK_CALL;
+
+static MOCK_CALL expected_calls[MAX_MOCK_CALL_COUNT];
+static size_t expected_call_count;
+static size_t actual_call_count;
+static bool actual_and_expected_match;
+
+int mock_snprintf(char* s, size_t n, const char* format, ...)
+{
+    int result;
+
+    if ((actual_call_count == expected_call_count) ||
+        (expected_calls[actual_call_count].mock_call_type != MOCK_CALL_TYPE_snprintf))
+    {
+        actual_and_expected_match = false;
+        return -1;
+    }
+    else
+    {
+        if (expected_calls[actual_call_count].u.snprintf_call.override_result)
+        {
+            result = expected_calls[actual_call_count].u.snprintf_call.call_result;
+        }
+        else
+        {
+            va_list args;
+            va_start(args, format);
+
+            expected_calls[actual_call_count].u.snprintf_call.format_arg = format;
+
+            result = vsnprintf(s, n, format, args);
+
+            va_end(args);
+        }
+
+        actual_call_count++;
+    }
+
+    return result;
+}
+
+int mock_vsprintf(char* s, const char* format, va_list args)
```

Comment:
in "log_context_property_type_ascii_char_potr_mocked.c" this is 

```c
extern int mock_vsprintf(char* s, const char* format, ...);
```

Here is has "va_list args".

Doesn't match for me. Which is wierd becayse the compiler seems to be oblivious to this.